### ISSUE TITLE
[Bug] Strange highlight of diff

### ISSUE THREAD

**Issue by sm-g** [ForkIssues/TrackerWin#94]
Fork: 
![](https://i.imgur.com/DiX5xNY.png)
VisualStudio / GitKraken: 
![](https://i.imgur.com/lA9krhY.png)

**Comment by Otiel** [comment]
This is always confusing and that makes me check twice if I didn't mess the first letter of the `Import` word.

**Comment by Otiel** [comment]
Here's another one:

![image](https://user-images.githubusercontent.com/3851902/54700802-fd80c200-4b33-11e9-8981-51978c0f7087.png)

Only the following text has been added at the end of the file:
```
## New release

[This template](new-release.md) defines the basic steps that should be followed when creating a new release. It should be copied to each repository and adapted with specific rules if needed.
```

**Comment by DanPristupov** [comment]
The line [7] used to be ended without `\n` symbol.

In the update you added `\n` to the end of [7] and also added some other lines.

The `\n` symbol is hidden but it also belongs to lines.

Update:

So, in your case the source was like following:

```
## Git workflow\n
\n
Repositories must follow the git workflow defined [ere](git-forkflow.ms).
```

And here's the new version:

```
## Git workflow\n
\n
Repositories must follow the git workflow defined [ere](git-forkflow.ms).\n
## New release\n
\n
[This template](new-release.md) defines the basic steps that should be followed when creating a new release. It should be copied to each repository and adapted with specific rules if needed.
```

The 'No newline at end of file' hint is important :).

**Comment by Otiel** [comment]
Fair enough for the `\n` character.

But why is the `No newline at end of file` displayed twice, and not only once at the end of the diff?

**Comment by DanPristupov** [comment]
Because `No newline at end of file` is related to the last line, not to the whole diff.

**Comment by jerone** [comment]
[**@DanPristupov**](https://github.com/DanPristupov) commented on [Mar 21, 2019, 10:55 AM GMT+1](https://github.com/ForkIssues/TrackerWin/issues/94#issuecomment-475168445 "2019-03-21T09:55:31Z - Replied by Github Reply Comments"):
> Because `No newline at end of file` is related to the last line, not to the whole diff.

It would be nice if that message could be replaced by something visual. Every time I see that message, I wonder if I wrote that 😊

**Comment by Otiel** [comment]
>Because `No newline at end of file` is related to the last line, not to the whole diff.

That makes sense. 👌 

>Fair enough for the `\n` character.

Coming back on this, is there any way you could visually highlight the `\n`? See for instance this diff:

![image](https://user-images.githubusercontent.com/3851902/54760736-f1a70580-4bf0-11e9-928d-fdd5184f0efa.png)

**Comment by DanPristupov** [comment]
The algorithmic part is not difficult, however I'm not sure if the AvalonEdit control supports such  rendering of `\n` symbol.

**Comment by DanPristupov** [comment]
The original issue fixed in 1.31
